ansible-dk-cli/ansible-dk

```diff
@@ -105,15 +136,54 @@ def verify():
 def generate():
     pass
 
-@generate.command(name='playbook')
-@click.option('--name', required=True, help='Name of playbook to generate')
-def generate_playbook(name):
-    print("Generating playbook: ", name)
+def parse_extra_vars(context, param, e):
+    extra_vars = {}
+    for var in e:
+        try:
+            key, value = var.split('=')
+        except ValueError:
+            raise click.BadParameter('must be in the format var=value')
+        extra_vars[key] = value
+    return extra_vars
+
+def generate_method(thing):
+    @generate.command(name=thing)
+    @click.option('--debug', is_flag=True, help="Enable debug mode")
+    @click.option('--generator', '-g', default=get_config()['generator'], help='Path to custom generator repo')
+    @click.option('-e', 'extra_vars', multiple=True, callback=parse_extra_vars, help='extra vars to pass to ansible, var=value')
+    @click.argument('name')
+    def generated_method(debug, generator, name, extra_vars):
```

Comment:
You must comment this section to describe your intent as to what is going on here.  Cleverness is harmful, remember this code has to be maintained by the whole team.